src/parser/index.ts

```diff
@@ -164,13 +164,24 @@ const removeTitle = (document: string) => {
 };
 
 /**
- * Cuts a string down to the specified length if needed
+ * Cuts a string down to the discord character limit and corrects missing codeblock backticks if needed
  * Appends '...' if the string was trimmed
+ * Appends '...```' if the codeblock backticks were corrected
  * @param {string} document
  * @param {number} length
  */
-const truncateString = (document: string, length: number) => {
-    return document.length > length ? document.slice(0, length - 3) + '...' : document;
+const truncateString = (document: string) => {
+    const MAX_LENGTH = 1024;
+    // Returns original document if its length is smaller than the discord maximum length
+    if (MAX_LENGTH > document.length) return document;
+    // Prepare document for truncate, appending ellipsis
+    const truncatedStr = document.slice(0, MAX_LENGTH - 3) + '...';
+    // Splits truncated string on \n and \r characters 
+    let lines = truncatedStr.match(/[^\r\n]+/g);
+    // Filter lines for codeblock backticks, store array length
+    let backtickMatches = lines?.filter(item => item === '```' || item === '```js').length;
```

Comment:
There are more codeblock syntaxes supported here. You can view a complete list at https://github.com/Thenlie/mdnman/blob/main/src/cli/output.ts#L8. Rather than accounting for all these languages individually with a big `||` chain, I think we can just check if the line `.startsWith('```')`

Reply:
Haven't seen `.startsWith()` before. Nice.